index.rst

```diff
@@ -233,3 +233,60 @@ The time spent serializing alerts into Avro format on the producer end dominates
    :name: figure-5
 
    Alert transit time vs volume of alerts.
+
+:numref:`figure-6` and :numref:`figure-7` show the average memory usage by Kafka over the length of each experiment.
+A back-of-the-envelope calculation for estimating memory needs says that if you want Kafka to buffer for 30 seconds then
+the memory need is write_throughput*30.  If we say on average the alert throughput is at least 250 alerts/second
+(but really bursty and likely higher) then the memory estimate is 250 alerts/sec * 135 KB/alert * 30 seconds = 10+ Gigabytes.
+Memory usage increases more rapidly with data volume for the larger single alert size (including postage stamps),
+nearing the maximum for the compute instance size for 1,000 and 10,000 alerts per visit with stamps.
+
+.. figure:: _static/memoryAlerts.png
+   :width: 55%
+   :align: center
+   :name: figure-6
+
+   Kafka memory usage vs number of alerts in a batch.
+
+.. figure:: _static/memorySize.png
+   :width: 55%
+   :align: center
+   :name: figure-7
+
```

Comment:
Do we know why the memory is saturating in the case with stamps?  Does this imply that the 10^4 measurement is swapping?  Naïvely, I'd think that should be essentially linear.

Reply:
No, I don't know.  This one I need to explore more.  It might be that the individual message sizes are large, and Kafka seems to move messages in batches of a certain size limit, amounting to ~7 messages per batch with the default setting.  Without stamps it can do ~3x as many in a batch.  Some sort of buffering difference is all I can guess now.